protocol/src/main/scala/io/mediachain/protocol/Datastore.scala

```diff
@@ -42,10 +42,22 @@ object Datastore {
     override val mediachainType: Option[MediachainType] = None
 
     override def toCbor: CValue =
-      CMap.withStringKeys("@link" -> CBytes(multihash.bytes))
+      CTaggedValue(MultihashReference.cborLinkTag,
+        CBytes(MultihashReference.multiaddrHeader ++ multihash.bytes))
   }
 
   object MultihashReference {
+    // IPLD uses cbor tags to indicate a traversable link
+    // if the tagged value is a byte array, it will be interpreted
+    // as a multiaddr (see https://github.com/jbenet/multiaddr)
+    val cborLinkTag: Long = 258
+
+    // prefix multihash bytes with a multiaddr header
+    // the header value is 0xA50322 which is the ipfs multiaddr code 0xA503
+    // (421, encoded as a protobuf-style varint)
+    // + the length of a sha2-256 multihash 0x22 (34 bytes)
+    val multiaddrHeader: Array[Byte] = Array[Byte](-91, 3, 34)
```

Comment:
The scala hex literals are interpreted as `Int`, which is kind of annoying, since e.g. `0xA5` is interpreted as outside the (signed) range of a `Byte`.  So you have to manually cast it to `Byte`:  

```
scala> 0xa5 
res0: Int = 165
scala> val b: Byte = 0xA5 
Main.scala:45: type mismatch;
 found   : Int(165)
 required: Byte
val b: Byte = 0xA5
              ^
Compilation Failed
scala> val b: Byte = 0xA5.asInstanceOf[Byte] 
b: Byte = -91
```